### PR TITLE
 Honda HRV

### PR DESCRIPTION
**Pull request by cowanhm**
values.py - incl. Class, FP, FW_VERSIONS, SPEED_FACTOR 1.025
carstate.py - incl. carFingerprint, doorOpen
Safety test verifies DoorOpen and/or seatbelt_unbuckle disengages OP
interface.py - incl. HR-V vehicle parameters, tuning (tested for 100+ miles)
Longitudinal tuning parameters unchanged from CAR.FIT

test_car_models.py - added per request of review process
created a side branch for these hrv specific updates to master

*5. honda_hrv_touring_2019_can.dbc has been submitted to commaai/opendbc and is called in values.py for CAR.HRV

Thanks, Cowan Moore 330-701-3922

### CODE REVIEW

**Comment by pd0wm** [comment]
The route is synced.

Are you 100% sure the door signals are not on can? Not disengaging on door open would be a safety issue.

**Comment by cowanhm** [comment]
We looked for several weeks.  The BO_ 1029 is not in this car's FP.  So, we setup a test with 35 open-close events.  Nothing in cabana.

@csouers helped me search further for the the signal. [Cabana Link](https://my.comma.ai/cabana/?route=03be5f2fd5c508d1%7C2020-04-16--15-06-29&exp=1618936539&sig=kHhbJIXUH50D4g8tC3by82BvTSQZiabcSUvYUgspZPY%3D&max=0&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F03be5f2fd5c508d1%2F2afd00bb18cc672b09d876124412763e_2020-04-16--15-06-29)

I've posted a test-drive video showing that seatbelt_unlatch disengages OP in openpilot_community_video yesterday @ 5:30.  That's the best I can provide at this point.

**Comment by cowanhm** [comment]
Well, your hunch was right - it does exist. After test against other honda vehicles, found the door signal matched the "honda_odyssey_extreme_edition_2018_china_can".  Updated carstate.py
BO_ 422 SCM_BUTTONS: 8 SCM
 SG_ DRIVERS_DOOR_OPEN : 63|1@0+ (1,0) [0|1] "" EON
Also updated the honda_hrv_touring_2019_can.dbc on the opendbc repositor to incl 422...63

**Comment by pd0wm** [comment]
You opendbc PR is merged. If you rebase your branch against master you should have the latest dbc.

**Comment by cowanhm** [comment]
Hope the rebase works...

File: selfdrive/car/honda/values.py [hunk 1]
```diff
@@ -681,6 +685,13 @@ class CAR:
       b'77959-TXM-A230\x00\x00',
     ],
   },
+  CAR.HRV: {
+    (Ecu.gateway, 0x18daeff1, None): [b'38897-T7A-A010\x00\x00'],
+    (Ecu.eps, 0x18da30f1, None): [b'39990-THX-A020\x00\x00'],
+    (Ecu.fwdRadar, 0x18dab0f1, None): [b'36161-T7A-A240\x00\x00'],
+    (Ecu.srs, 0x18da53f1, None): [b'77959-T7A-A230\x00\x00'],
+    (Ecu.combinationMeter, 0x18da60f1, None): [b'78109-THX-A210\x00\x00'],
```
Comment (pd0wm) [inline selfdrive/car/honda/values.py:693]: No fwdCamera? Are you sure this is the complete list?

**Comment by pd0wm** [comment]
Also add the HRV to `IGNORED_FINGERPRINTS`, and set `fingerprintSource: 'fixed'` in test_car_models. We're only doing fingerprinting 2.0 for new Honda/Toyota models.

File: selfdrive/car/honda/values.py (hunk 1 above)
Comment (cowanhm) [reply selfdrive/car/honda/values.py:693]: these were all the codes under carFw that csouers instructed us.  The car does have a forward camera, but not sure why its not showing up.  When c2 is powered off, the camera works fine with honda lkas.
  
ecu2

**Comment by cowanhm** [comment]
Changes make per you instruction

**Comment by cowanhm** [comment]
Thx pd0wm- my first "all checks have passed".  Time to open a corona.

**Comment by cowanhm** [comment]
Thx pd0wm - my first "all checks have passed".

**Comment by geohot** [comment]
Add car to README?

**Comment by cowanhm** [comment]
Sure.  Loaded master fresh, did a test drive... All good!

**Pull request merged**